scam-detector-py/src/blockchain_indexer_service.py

```diff
@@ -225,6 +225,37 @@ def has_deployed_high_tx_count_contract(address, chain_id) -> bool:
                         break
                     time.sleep(1)
         return False
+
+    @staticmethod
+    @RateLimiter(max_calls=1, period=1)
+    def get_contract_nonce(address, block_number, chain_id) -> int:
+        logging.info(f"get_contract_nonce for contract {address} on {chain_id} called.")
+
+        transactions_for_contract = f"{BlockChainIndexer.get_etherscan_url(chain_id)}/api?module=account&action=txlistinternal&address={address}&startblock=0&endblock={block_number}&page=1&offset=10000&sort=asc&apikey={BlockChainIndexer.get_api_key(chain_id)}"
+
+        contract_nonce = 0
+        success = False
+        count = 0
+        while not success:
+            data = requests.get(transactions_for_contract)
+            if data.status_code == 200:
+                json_data = json.loads(data.content)
+                success = True
+                if isinstance(json_data["result"], list):
+                            # Block explorer API returns up to 10000 results
+                            if len(json_data["result"]) == 10000:
+                                return 0
+                            for result in json_data["result"]:
+                                if result["contractAddress"] != "" and result["isError"] == "0":
+                                    contract_nonce += 1
```

Comment:
when is the nonce incremented on chain?

Reply:
You're right, I updated it to use the on-chain nonce.